tests/unit/cluster/scripting.tcl

```diff
@@ -62,6 +62,10 @@ start_cluster 1 0 {tags {external:skip cluster}} {
             } 1 bar}
     }
 
+    test {Cross slot commands are allowed by default if they disagree with pre-declared keys} {
+        r 0 eval "redis.call('set', 'foo', 'bar')" 1 bar
```

Comment:
could that test have detected the bug?
don't we need another (non-script) command to try to read that key and make sure it reached the right dict?

Reply:
we have an assert inside `getKeySlot`, and our CI make with `DEBUG_ASSERTIONS`, so this case can detect.

```c
int getKeySlot(sds key) {
    /* This is performance optimization that uses pre-set slot id from the current command,
     * in order to avoid calculation of the key hash.
     * This optimization is only used when current_client flag `CLIENT_EXECUTING_COMMAND` is set.
     * It only gets set during the execution of command under `call` method. Other flows requesting
     * the key slot would fallback to calculateKeySlot.
     */
    if (server.current_client && server.current_client->slot >= 0 && server.current_client->flags & CLIENT_EXECUTING_COMMAND) {
        debugServerAssertWithInfo(server.current_client, NULL, calculateKeySlot(key)==server.current_client->slot);
        return server.current_client->slot;
    }
    return calculateKeySlot(key);
}
```

Reply:
Glad we added that assert.

Reply:
still, i think it's a good idea to add a line to the test to try to GET the key and see it's in the right place.
that may be true for a few other script tests.
maybe some day the code will be refactored and the assertions won't be triggered.